internal/handlers/gateway/modes_test.go

```diff
@@ -0,0 +1,74 @@
+package gateway
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/os-observability/tempo-operator/apis/tempo/v1alpha1"
+)
+
+func TestValidateModes(t *testing.T) {
+	tt := []struct {
+		name    string
+		input   v1alpha1.TempoStack
+		wantErr string
+	}{
+		{
+			name: "another mode",
+			input: v1alpha1.TempoStack{
+				Spec: v1alpha1.TempoStackSpec{
+					Tenants: &v1alpha1.TenantsSpec{},
+				},
+			},
+		},
+		{
+			name: "static missing authentication",
+			input: v1alpha1.TempoStack{
+				Spec: v1alpha1.TempoStackSpec{
+					Tenants: &v1alpha1.TenantsSpec{
+						Mode: v1alpha1.Static,
+					},
+				},
+			},
+			wantErr: "mandatory configuration - missing tenants' authentication configuration",
+		},
+		{
+			name: "static missing roles",
+			input: v1alpha1.TempoStack{
+				Spec: v1alpha1.TempoStackSpec{
+					Tenants: &v1alpha1.TenantsSpec{
+						Mode:           v1alpha1.Static,
+						Authorization:  &v1alpha1.AuthorizationSpec{},
+						Authentication: []v1alpha1.AuthenticationSpec{},
+					},
+				},
+			},
+			wantErr: "mandatory configuration - missing roles configuration",
+		},
+		{
+			name: "static missing role bindings",
+			input: v1alpha1.TempoStack{
+				Spec: v1alpha1.TempoStackSpec{
+					Tenants: &v1alpha1.TenantsSpec{
+						Mode: v1alpha1.Static,
+						Authorization: &v1alpha1.AuthorizationSpec{
+							Roles: []v1alpha1.RoleSpec{},
+						},
+						Authentication: []v1alpha1.AuthenticationSpec{},
+					},
+				},
+			},
+			wantErr: "mandatory configuration - missing role bindings configuration",
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			err := ValidateModes(tc.input)
+			if tc.wantErr != "" {
```

Comment:
The else branch should check for nil error. 

Maybe this can be written without branching and directly check the value.